apps/arweave/src/ar_mining_server.erl

```diff
@@ -691,38 +685,14 @@ prepare_solution(poa1, Candidate, Solution) ->
 									{modules_covering_recall_byte, ModuleIDs}]),
 							error
 					end;
-				{Chunk1, {error, Error}} ->
-					Modules = ar_storage_module:get_all(RecallByte1 + 1),
-					ModuleIDs = [ar_storage_module:id(Module) || Module <- Modules],
-					?LOG_ERROR([{event, failed_to_find_poa_proofs_locally},
-							{error, io_lib:format("~p", [Error])},
-							{tags, [solution_proofs]},
-							{recall_byte, RecallByte1},
-							{packing_difficulty, PackingDifficulty},
-							{modules_covering_recall_byte, ModuleIDs}]),
-					ar:console("WARNING: we have mined a block but did not find the PoA1 proofs "
-							"locally - searching the peers...~n"),
-					case fetch_poa_from_peers(RecallByte1, PackingDifficulty) of
```

Comment:
Why do we remove fetch_poa_from_peers from here?

Reply:
it all happens in prepare_poa now up here: https://github.com/ArweaveTeam/arweave/pull/628/files#diff-ea8e957c63acb7497b0ed4f9201b40f18e5d5b4fa371e67b4cff05850e4df854R646

We had 3 spots where we potentially needed to read or fetch poa when preparing a solution poa1, preparing a solution poa2, and before sending an H2 candidate to a CM peer. Previously the 3rd case was missing the read and fetch. So the intent was to pull the logic together into a single function (prepare_poa) and then call that function from each of the 3 spots that need to read/fetch the poa.

that was the intent at least...

Reply:
Ah, yes, I see, thank you